chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -78,17 +79,36 @@
         node[:crowbar_wall][:boot_device] = "disk/by-id/#{bootdisk}"
       end
     end
+
+    # Searching boot disk with virtio driver if no other disks were found
+    # on SLES_12 sleshammer image
+    if dev.nil?
+      basedir = "/sys/block"
+      ::Dir.entries(basedir).sort.each do |path|
+        next unless File.symlink?(File.join(basedir, path))
+        dev = File.readlink("#{basedir}/#{path}").split("/")[-1]
+        if dev =~ /^vd[a-z]+$/
```

Comment:
thats kind of pointless. the symlink will always point to the same name. you're iterating sorted over /sys/block/..

how about you're doing the sorted iteration, but simply check if the link contains "/virtio" ? 

in a way it feels like you actually want to sort by the link, not by the name of the link..